backend/src/data_entry/mod.rs

```diff
@@ -65,12 +67,12 @@ pub async fn polling_station_data_entry_save(
     State(polling_station_data_entries): State<PollingStationDataEntries>,
     State(polling_stations_repo): State<PollingStations>,
     State(elections): State<Elections>,
-    Path((id, entry_number)): Path<(u32, u8)>,
+    Path((id, entry_number)): Path<(u32, EntryNumber)>,
     data_entry_request: SaveDataEntryRequest,
 ) -> Result<SaveDataEntryResponse, APIError> {
     // Check if it is valid to save the data entry
     match entry_number {
```

Comment:
It doesn't look like we have tests for this?

Reply:
True and they already should have been there. Only the happy path is tested. 
This `match` will be refactored in the state machine PR that is in progress now, so I'm inclined to add the tests when the refactor is done.

Reply:
Works for me. Am guessing it does not make sense to add some tests for the new enum itself?

Reply:
I'd say it does makes sense and we have them by calling `EntryNumber::try_from()` in the tests. Serialization and Encoding paths are tested via integration- and DOM2DB tests. But again, we currently only test happy paths. I'll make a note in the issue to make sure we test error cases.

Reply:
Added a sub issue: https://github.com/kiesraad/abacus/issues/728